src/com/palmergames/bukkit/towny/db/TownyFlatFileSource.java

```diff
@@ -1349,6 +1397,109 @@ public boolean loadWorld(TownyWorld world) {
 			return false;
 		}
 	}
+	/*
+	public boolean loadGroups() {
+		String line = "";
+		String path;
+		
+		for (PlotGroup group : getAllGroups()) {
+			path = getGroupFilename(group);
+			
+			File groupFile = new File(path);
+			
+			if (groupFile.exists() && groupFile.isFile()) {
+				String test = null;
+
+				try {
+					HashMap<String, String> keys = new HashMap<>();
+					Properties properties = new Properties();
+					properties.load(new InputStreamReader(new FileInputStream(groupFile), StandardCharsets.UTF_8));
+					for (String key : properties.stringPropertyNames()) {
+						String value = properties.getProperty(key);
+						keys.put(key, String.valueOf(value));
+					}
+					
+					line = keys.get("name");
+				}
+				
+			}
+		}
+	}
+	
+	 */
+	@Override
+	public boolean loadPlotGroups() {
+		String line = "";
+		String path;
+		
+		TownyMessaging.sendErrorMsg("" + getAllGroups().size());
+		
+		for (PlotGroup group : getAllGroups()) {
+			path = getGroupFilename(group);
+			
+			File groupFile = new File(path);
+			if (groupFile.exists() && groupFile.isFile()) {
+				String test = null;
+				try {
+					HashMap<String, String> keys = new HashMap<>();
+					Properties properties = new Properties();
+					properties.load(new InputStreamReader(new FileInputStream(groupFile), StandardCharsets.UTF_8));
+					for (String key : properties.stringPropertyNames()) {
+						String value = properties.getProperty(key);
+						keys.put(key, String.valueOf(value));
+					}
+					
+					line = keys.get("groupName");
+					if (line != null)
+						group.setGroupName(line.trim());
+					
+					line = keys.get("groupID");
+					if (line != null)
+						group.setID(Integer.parseInt(line.trim()));
+					
+					test = "town";
+					line = keys.get("town");
+					if (line != null && !line.isEmpty()) {
+						Town town = getTown(line.trim());
+						group.setTown(town);
+					}
+					else {
+						TownyMessaging.sendErrorMsg("Could not add to town!");
+						deleteGroup(group);
+					}
+					
+					line = keys.get("groupPrice");
+					if (line != null && !line.isEmpty())
+						group.setPrice(Double.parseDouble(line.trim()));
+					
+					line = keys.get("resident");
+					if (line != null && !line.isEmpty())
+						group.setResident(getResident(line.trim()));
+						
+					
+				} catch (Exception e) {
+					if (test.equals("town")) {
+						TownyMessaging.sendDebugMsg("Group file missing Town, deleting " + path);
+						deleteGroup(group);
+						TownyMessaging.sendDebugMsg("Missing file: " + path + " deleting entry in group.txt");
+						//TownyWorld world = townBlock.getWorld();
+						//world.removeTownBlock(townBlock);
```

Comment:
Have this so it removes the Plot group from the global list if it's town cannot be found.